sw/device/silicon_creator/lib/manifest.h

```diff
@@ -19,6 +19,18 @@
 extern "C" {
 #endif  // __cplusplus
 
+// Extern declarations for code_start, code_end, and entry_point in the
+// manifest.
+extern const uint32_t _text_start;
+extern const uint32_t _text_end;
+
+#define MANIFEST_DECLARE(entrypoint_) \
+  const volatile manifest_t kManifest __attribute__((section (".manifest"))) = { \
+    .code_start = (uintptr_t)(&_text_start), \
+    .code_end = (uintptr_t)(&_text_end), \
+    .entry_point = entrypoint_, \
```

Comment:
These fields are offsets relative to the start of the [manifest](https://docs.opentitan.org/sw/device/silicon_creator/rom_ext/docs/manifest/). Shouldn't we subtract the address of the manifest for this to work for rom_exts built for slot b or address translation, and the first owner boot stage? We should also confirm that the values match after this change.

Also, I don't think bare_metal_common.ld has `_text_end` defined yet.